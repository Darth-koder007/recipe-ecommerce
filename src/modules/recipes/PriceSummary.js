import List, { ListItem } from '../../components/List';

import Flex from '../../components/Flex';
import PropTypes from 'prop-types';
import React from 'react';
import Text from '../../components/Text';
import { parseRawPrice } from './price';

const PriceSummaryItem = ({ name, price, selected }) => (
  <ListItem>
    <Flex paddingBottom="xs" justifyContent="space-between">
      <Text as="span">
        {name}
        {!!selected && selected > 1 && ` x ${selected}`}
      </Text>
      <Text as="span">{parseRawPrice(price)}</Text>
    </Flex>
  </ListItem>
);

PriceSummaryItem.propTypes = {
  id: PropTypes.string,
  name: PropTypes.string,
  price: PropTypes.number,
  selected: PropTypes.number,
};

// PriceSummary user interface
const EmptyPriceSummary = () => (
  <Text fontWeight="bold" as="p" textAlign="center">
    Your Box Is Currently Empty
  </Text>
);

// PriceSummary user interface
const PriceSummary = ({ summary, totalPrice }) => (
  <List width={['290px', '450px']} padding="sm" data-testid="price-summary">
    {summary.map(({ id, ...restSummary }) => (
      <PriceSummaryItem key={id} {...restSummary} />
    ))}
    {summary.length > 0 ? (
      <ListItem>
        <Flex
          borderTopWidth="sm"
          borderTopColor="border"
          borderTopStyle="solid"
          justifyContent="space-between"
          alignItems="center"
          paddingTop="xs">
          <Text fontWeight="bold" as="span">
            Total
          </Text>
          <Text fontWeight="bold" as="span">
            {parseRawPrice(totalPrice)}
          </Text>
        </Flex>
      </ListItem>
    ) : (
      <EmptyPriceSummary />
    )}
  </List>
);

PriceSummary.propTypes = {
  totalPrice: PropTypes.number,
  summary: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.string,
      name: PropTypes.string,
      price: PropTypes.number,
      selected: PropTypes.number,
    })
  ),
};

export default PriceSummary;
