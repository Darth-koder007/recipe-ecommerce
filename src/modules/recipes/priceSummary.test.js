import { render, screen } from '@testing-library/react';

import PriceSummary from './PriceSummary';
import React from 'react';
import { parseRawPrice } from './price';
import priceSummaryFixtures from './priceSummary.fixtures';

describe('Price summary', () => {
  test('should match all the items present in summary', () => {
    // explicitly stating this so that jest knows how many assertions will be there
    expect.assertions(priceSummaryFixtures.testMultiQuantityData.summary.length + 1);
    render(<PriceSummary {...priceSummaryFixtures.testMultiQuantityData} />);

    const listItems = screen.getAllByRole('listitem');

    listItems.forEach((item, index) => {
      const dataItem = priceSummaryFixtures.testMultiQuantityData.summary[index];
      let text;

      if (index === priceSummaryFixtures.testMultiQuantityData.summary.length) {
        // for testing total price containing list item
        text = `Total${parseRawPrice(priceSummaryFixtures.testMultiQuantityData.totalPrice)}`;
      } else {
        // for testing recipe summary item
        text = `${dataItem.name}${
          !!dataItem.selected && dataItem.selected > 1 ? ` x ${dataItem.selected}` : ``
        }${parseRawPrice(dataItem.price)}`;
      }

      expect(item).toHaveTextContent(text);
    });
  });

  test('should render empty box message', () => {
    render(<PriceSummary {...priceSummaryFixtures.testEmptyData} />);

    expect(screen.getByText('Your Box Is Currently Empty')).toBeVisible();
  });

  test('should match snapshot', () => {
    expect(
      render(<PriceSummary {...priceSummaryFixtures.testMultiQuantityData} />)
    ).toMatchSnapshot();
  });
});
