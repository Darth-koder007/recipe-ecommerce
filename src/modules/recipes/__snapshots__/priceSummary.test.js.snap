// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Price summary should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <ul
        class="sc-gtssRu deunSs"
        data-testid="price-summary"
        width="290px,450px"
      >
        <li
          class="sc-bdnylx"
        >
          <div
            class="sc-dlnjPT sc-hKFyIo fTRecQ"
            display="flex"
          >
            <span
              class="sc-eCApGN iDQzKM"
            >
              Chicken Sausage & Spinach Ravioli
            </span>
            <span
              class="sc-eCApGN iDQzKM"
            >
              $17.98
            </span>
          </div>
        </li>
        <li
          class="sc-bdnylx"
        >
          <div
            class="sc-dlnjPT sc-hKFyIo fTRecQ"
            display="flex"
          >
            <span
              class="sc-eCApGN iDQzKM"
            >
              Gouda Vibes Burgers
               x 2
            </span>
            <span
              class="sc-eCApGN iDQzKM"
            >
              $35.96
            </span>
          </div>
        </li>
        <li
          class="sc-bdnylx"
        >
          <div
            class="sc-dlnjPT sc-hKFyIo fTRecQ"
            display="flex"
          >
            <span
              class="sc-eCApGN iDQzKM"
            >
              Figgy Balsamic Pork
            </span>
            <span
              class="sc-eCApGN iDQzKM"
            >
              $17.98
            </span>
          </div>
        </li>
        <li
          class="sc-bdnylx"
        >
          <div
            class="sc-dlnjPT sc-hKFyIo fTRecQ"
            display="flex"
          >
            <span
              class="sc-eCApGN iDQzKM"
            >
              Duck a l’Orange
            </span>
            <span
              class="sc-eCApGN iDQzKM"
            >
              $29.96
            </span>
          </div>
        </li>
        <li
          class="sc-bdnylx"
        >
          <div
            class="sc-dlnjPT sc-hKFyIo fTRecQ"
            display="flex"
          >
            <span
              class="sc-eCApGN iDQzKM"
            >
              Shipping
            </span>
            <span
              class="sc-eCApGN iDQzKM"
            >
              $12.98
            </span>
          </div>
        </li>
        <li
          class="sc-bdnylx"
        >
          <div
            class="sc-dlnjPT sc-hKFyIo eNIeVt"
            display="flex"
          >
            <span
              class="sc-eCApGN fLyelG"
              font-weight="bold"
            >
              Total
            </span>
            <span
              class="sc-eCApGN fLyelG"
              font-weight="bold"
            >
              $114.86
            </span>
          </div>
        </li>
      </ul>
    </div>
  </body>,
  "container": <div>
    <ul
      class="sc-gtssRu deunSs"
      data-testid="price-summary"
      width="290px,450px"
    >
      <li
        class="sc-bdnylx"
      >
        <div
          class="sc-dlnjPT sc-hKFyIo fTRecQ"
          display="flex"
        >
          <span
            class="sc-eCApGN iDQzKM"
          >
            Chicken Sausage & Spinach Ravioli
          </span>
          <span
            class="sc-eCApGN iDQzKM"
          >
            $17.98
          </span>
        </div>
      </li>
      <li
        class="sc-bdnylx"
      >
        <div
          class="sc-dlnjPT sc-hKFyIo fTRecQ"
          display="flex"
        >
          <span
            class="sc-eCApGN iDQzKM"
          >
            Gouda Vibes Burgers
             x 2
          </span>
          <span
            class="sc-eCApGN iDQzKM"
          >
            $35.96
          </span>
        </div>
      </li>
      <li
        class="sc-bdnylx"
      >
        <div
          class="sc-dlnjPT sc-hKFyIo fTRecQ"
          display="flex"
        >
          <span
            class="sc-eCApGN iDQzKM"
          >
            Figgy Balsamic Pork
          </span>
          <span
            class="sc-eCApGN iDQzKM"
          >
            $17.98
          </span>
        </div>
      </li>
      <li
        class="sc-bdnylx"
      >
        <div
          class="sc-dlnjPT sc-hKFyIo fTRecQ"
          display="flex"
        >
          <span
            class="sc-eCApGN iDQzKM"
          >
            Duck a l’Orange
          </span>
          <span
            class="sc-eCApGN iDQzKM"
          >
            $29.96
          </span>
        </div>
      </li>
      <li
        class="sc-bdnylx"
      >
        <div
          class="sc-dlnjPT sc-hKFyIo fTRecQ"
          display="flex"
        >
          <span
            class="sc-eCApGN iDQzKM"
          >
            Shipping
          </span>
          <span
            class="sc-eCApGN iDQzKM"
          >
            $12.98
          </span>
        </div>
      </li>
      <li
        class="sc-bdnylx"
      >
        <div
          class="sc-dlnjPT sc-hKFyIo eNIeVt"
          display="flex"
        >
          <span
            class="sc-eCApGN fLyelG"
            font-weight="bold"
          >
            Total
          </span>
          <span
            class="sc-eCApGN fLyelG"
            font-weight="bold"
          >
            $114.86
          </span>
        </div>
      </li>
    </ul>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
